### PR TITLE
qemu-guest-agent: windows guest support for gagent snapshot test case 

### PR DESCRIPTION
**Pull request by xiagao**
1. windows guest support for qemu_guest_agent_snapshot
2. delete redundant parameter for gagent snapshot
3. enlarge the transfer data size
ID: 1543267 1508302
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
(1/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_snapshot: PASS (82.17 s)
 (2/2) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_snapshot: PASS (402.86 s)

File: qemu/tests/qemu_guest_agent_snapshot.py [hunk 1]
```diff
@@ -55,18 +55,25 @@ def _action_after_fsthaw(self, *args):
...
+        finally:
+            error_context.context("Delete the created files.", logging.info)
+            process.run("rm -rf %s %s" % (self.host_path, self.host_path_returned))
+            session = self._get_session(self.params, None)
+            self._open_session_list.append(session)
+            cmd_del_file = "rm -rf %s" % self.guest_path
+            if self.params.get("os_type") == "windows":
+                cmd_del_file = r"del /f /q %s" % self.guest_path
+            session.cmd(cmd_del_file)
+            session.close()
```
Comment (hereischen) [inline qemu/tests/qemu_guest_agent_snapshot.py]: If you put the `session` into `self._open_session_list`, then you don't need to close it on line 76.
Or you don't put it into the list.

Comment (xiagao) [reply qemu/tests/qemu_guest_agent_snapshot.py]: You are right, delete it in L76. Pls have a check,thanks.

**Comment by xiagao** [comment]
@xiangchunfu  could you help review this patch?

File: qemu/tests/qemu_guest_agent_snapshot.py [hunk 2]
```diff
@@ -55,18 +55,24 @@ def _action_after_fsthaw(self, *args):
...
-        self.vm.destroy()
+        try:
+            self.host_path_returned = "%s-returned" % self.host_path
+            self.vm.copy_files_from(self.guest_path, self.host_path_returned)
+            error_context.context("comparing hashes", logging.info)
+            self.curr_hash = crypto.hash_file(self.host_path_returned)
+            if self.orig_hash != self.curr_hash:
+                self.test.fail("Current file hash (%s) differs from "
+                               "original one (%s)" % (self.curr_hash,
+                                                      self.orig_hash))
```
Comment (xiangchunfu) [inline qemu/tests/qemu_guest_agent_snapshot.py]: @xiagao I don't find reboot and shutdown guest in polarion.  But your test steps included this action.  If don't need this test.  please remove this item. or add this step to polarion. 8) Reboot and shutdown guest.

Comment (xiagao) [reply qemu/tests/qemu_guest_agent_snapshot.py]: @xfu check from polarion,there is no this step, delete it in docstring.
Pls have a check.
Thanks.

File: qemu/tests/qemu_guest_agent_snapshot.py [hunk 3]
```diff
@@ -55,18 +55,24 @@ def _action_after_fsthaw(self, *args):
...
-
-        error_context.context("Reboot and shutdown guest.")
-        self.vm.reboot()
-        self.vm.destroy()
+        try:
+            self.host_path_returned = "%s-returned" % self.host_path
+            self.vm.copy_files_from(self.guest_path, self.host_path_returned)
+            error_context.context("comparing hashes", logging.info)
+            self.curr_hash = crypto.hash_file(self.host_path_returned)
+            if self.orig_hash != self.curr_hash:
```
Comment (xiangchunfu) [inline qemu/tests/qemu_guest_agent_snapshot.py]: @xiagao line 34,  can bg.start() make sure hash value is correct after transferring data from host to guest?  If the first time is wrong, then this time  must orig_hash != self.curr_hash.

Comment (xiagao) [reply qemu/tests/qemu_guest_agent_snapshot.py]: @xiangchunfu If the file is changed during transfer, it's a product bug.
The steps is:
1) create a file in host. -original file in host
2) while transfering data do fsfreeze operation.
3) after thaw and data transfer finished, copy the file from guest to host. -the current file.
4) compare the two files hash value in host.

Comment (xiangchunfu) [reply qemu/tests/qemu_guest_agent_snapshot.py]: For function testing,  looks good to me. 
Acked-by: xfu@redhat.com

**Comment by xiagao** [comment]
@yanan-fu Could you also review this patch? Thank  you.

**Comment by xiagao** [comment]
@vivianQizhu  Could you help review this patch? thanks..

**Comment by vivianQizhu** [comment]
@xiagao Code looks good, would you provide test result please? For both linux and windows guest.

**Comment by xiagao** [comment]
@vivianQizhu  thanks.
 (1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.7.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_snapshot.q35: PASS (77.87 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_snapshot.q35: PASS (108.07 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Pull request merged**